File: src/Systems/Fortran/patterns/service_locator.f90
module service_locator_example
  implicit none
contains
  logical function run()
    run = trim(locate('email','a@example.test')) == 'email>a@example.test' .and. &
          trim(locate('audit','created')) == 'audit>created'
  contains
    function locate(service, value) result(text)
      character(len=*), intent(in) :: service, value
      character(len=96) :: text
      text = trim(service)//'>'//trim(value)
    end function
  end function
end module

File: src/Systems/Swift/patterns/ServiceLocator.swift
enum ServiceLocatorExample {
    static func run() -> Bool {
        let services: [String: (String) -> String] = [
            "email": { "email>\($0)" },
            "audit": { "audit>\($0)" }
        ]
        return services["email"]!("a@example.test") == "email>a@example.test"
            && services["audit"]!("created") == "audit>created"
    }
}
